src/shims/unix/fs.rs

```diff
@@ -605,12 +605,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
             // (Technically we do not support *not* setting this flag, but we ignore that.)
             mirror |= o_cloexec;
         }
-        let o_tmpfile = this.eval_libc_i32("O_TMPFILE")?;
-        if flag & o_tmpfile != 0 {
-            // if the flag contains `O_TMPFILE` then we return a graceful error
-            let eopnotsupp = this.eval_libc("EOPNOTSUPP")?;
-            this.set_last_error(eopnotsupp)?;
-            return Ok(-1);
+        if cfg!(target_os = "linux") {
```

Comment:
In the `src` folder, this checks the compile-time target, which is wrong. You need to check the run-time target. You can find that in `this.tcx.sess.target.os`.